### DISCUSSION TITLE
Is it possible to modify the navigation templates in the html format?

### DISCUSSION THREAD

**Discussion by dweng0** [quarto-dev/quarto-cli#10794] [Q&A]
### Description

**TL;DR** is there a way to modify the html templates to provide a different structure?

I am trying to customise the structure of the navigation bar so that I can have a slightly more unique style:
![image](https://github.com/user-attachments/assets/79fd6724-7806-45ff-b887-91af6291ee59)
I noticed that you can use ejs templates for lists. But couldn't find anything similar for the navbar/LHS nav.

However I was unable to find anything in the docs that points to how to setup an ejs template or modify a base navigation html template. I copied in the base html template and noticed that the navigation and let hand side navigation all come in via:

```html 
<body>
    <!-- header and LHS Nav start -->
    $for(include-before)$
    $include-before$
    $endfor$
    <!-- header end -->
``` 

but looking at the [reference](https://quarto.org/docs/reference/projects/websites.html) list in the `_quarto.yaml` file, I can't see anyway to modify that specific include-before.

**Comment by mcanouil** [comment]
No, there is not currently, but you can achieve a lot with (S)CSS.

See <https://realworlddatascience.net/>.
